File: .net/dojos/dojo1/Bowling/BowlingTest/LastFrameTest.cs
using System;
using Bowling;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace BowlingTest
{
    [TestClass]
    public class LastFrameTest
    {
        [TestMethod]
        public void LastFrameShouldJustAddThreeTries()
        {
            //given
            LastFrame lastFrame=new LastFrame {FirstTry = 2,SecondTry = 4,ThirdTry=6};

            //when
            int score=lastFrame.Score;

            //then
            Assert.AreEqual(12,score);
        }
    }
}

File: dotnet/dojos/dojo1/JBG/src/test/java/bowling/LastFrameTest.java
package bowling;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class LastFrameTest {
    @Test
    public void shouldOnlyCountTwoRollsWhenThirdNotPresent() {
        //given
        LastFrame lastFrame = new LastFrame(3, 2, 0);

        //when
        int score = lastFrame.countScore();

        //then
        assertThat(score, is(5));
    }

    @Test
    public void shouldOnlyCountOwnThreeRolls() {
        //given
        LastFrame lastFrame = new LastFrame(1, 9, 3);

        //when
        int score = lastFrame.countScore();

        //then
        assertThat(score, is(13));
    }

    @Test
    public void shouldNotHaveBonusWhenStrike() {
        //given
        LastFrame lastFrame = new LastFrame(10, 8, 3);

        //when
        int score = lastFrame.countScore();

        //then
        assertThat(score, is(21));
    }

}
